app/mailers/publisher_mailer.rb

```diff
@@ -2,19 +2,27 @@ class PublisherMailer < ApplicationMailer
   include PublishersHelper
   add_template_helper(PublishersHelper)
 
-  def login_email(publisher)
+  # Note: token_fresh is a reminder that you must refresh the token before sending this email.
+  # Best practice is to use the MailerServices::PublisherLoginLinkEmailer service
+  def login_email(publisher, token_fresh = false)
+    raise "token not fresh" unless token_fresh
```

Comment:
I like how this restricts the the use of token generation to only the MailerServices .  What do you think of also checking that the token exists and hasn't expired to be extra sure?

```ruby
raise if publisher.authentication_token == nil || publisher.authentication_token_expires_at <= Time.now
```

Reply:
I'll add the before filters and remove `token_fresh`.